src/synthesizer/extensions/particle_spectra.c

```diff
@@ -61,7 +61,7 @@ void spectra_loop_cic(const double **grid_props, const double **part_props,
 
       /* Use the grid edge. */
       part_cell = 0;
-      frac = 1;
```

Comment:
Why are we not using the CIC and NGP methods in `weights.h` here? All the code appears to be identical up to the weight summation

Reply:
You can't use that code here. This is per particle so you can't make a single grid of weights. If you made a grid of weights per particle the memory footprint would be prohibitive. That's why this is bespoke and loops over the spectra for each particle.